src/include/miopen/glu/problem_description.hpp

```diff
@@ -0,0 +1,240 @@
+        }
+
+        if(inputDesc.GetLengths()[dim] % 2 != 0)
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "GLU::ProblemDescription: The split dimension size of input tensor should "
+                         "be divisible by 2.");
+        }
+
+        for(auto i = 0; i < inputDesc.GetNumDims(); i++)
+        {
+            if(i == dim)
+            {
+                if(inputDesc.GetLengths()[i] / 2 != outputGradDesc.GetLengths()[i] ||
+                   inputDesc.GetLengths()[i] != inputGradDesc.GetLengths()[i])
+                {
+                    MIOPEN_THROW(miopenStatusBadParm,
+                                 "GLU::ProblemDescription: Dimension sizes don't match between "
+                                 "input tensor and output tensor.");
+                }
+            }
+            else
+            {
+                if(inputDesc.GetLengths()[i] != inputGradDesc.GetLengths()[i] ||
+                   inputDesc.GetLengths()[i] != outputGradDesc.GetLengths()[i])
+                {
+                    MIOPEN_THROW(miopenStatusBadParm,
+                                 "GLU::ProblemDescription: Dimension sizes don't match between "
+                                 "input tensor and output tensor.");
+                }
+            }
+        }
+
+        if(!IsSameType())
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "GLU::ProblemDescription: Tensor types do not match.");
+        }
+    }
+
+    Direction GetDirection() const { return direction; }
+    const TensorDescriptor& GetInputDesc() const { return inputDesc; }
+    const TensorDescriptor& GetInputGradDesc() const { return inputGradDesc; }
+    const TensorDescriptor& GetOutputDesc() const { return outputDesc; }
+    const TensorDescriptor& GetOutputGradDesc() const { return outputGradDesc; }
+    uint32_t GetDim() const { return dim; }
+
+    bool IsSameType() const
+    {
+        if(direction == Direction::Forward)
+        {
+            if(inputDesc.GetType() != outputDesc.GetType())
+            {
+                return false;
+            }
+        }
+        else
+        {
+            if(inputDesc.GetType() != inputGradDesc.GetType() ||
+               inputGradDesc.GetType() != outputGradDesc.GetType())
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    bool IsAllContiguous() const
+    {
+        if(direction == Direction::Forward)
+        {
+            if(!(inputDesc.IsContiguous() && outputDesc.IsContiguous()))
+            {
+                return false;
+            }
+        }
+        else
+        {
+            if(!(inputDesc.IsContiguous() && inputGradDesc.IsContiguous() &&
+                 outputGradDesc.IsContiguous()))
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    bool IsFirstDim() const
```

Comment:
IsFirstDim is a bit misleading. Probably it is better to add getter for `dim` and check `dim` value in place.